services/wireguard/service/service_windows.go

```diff
@@ -38,9 +39,16 @@ func NewManager(
 	location location.ServiceLocationInfo,
 	natService nat.NATService,
 	portMap func(port int) (releasePortMapping func()),
-	options Options) *Manager {
+	options Options,
+	portPool portSupplier,
+) *Manager {
 
-	resourceAllocator := resources.NewAllocator(options.PortMin, options.Subnet)
+	portSupplier := portPool
+	if options.PortMin != 0 {
```

Comment:
we should probably just accept a portSupplier interface as a param. We then don't have to perform this check here and can just consume the port supplier.

Reply:
The idea was to encapsulate `PortPool` as a `portSupplier` (inline with openvpn implementation). So the service itself doesn't need to care about whether it's using a random or a fixed port (supplier can be either `port.Pool` or `port.Fixed`). WDYT?

Reply:
That's basicly my thinking, except I'd define an interface here and move the check you're performing even further up. That way the wireguard service itself does not need to know anything about the port package.

Reply:
On the second thought, the highest in the hierarchy where the supplier can be configured is `Dependencies.bootstrapServiceWireguard`, because openvpn and wireguard use separate flags, thus it must be configured individually for a specific service. And this kind of functionality does belong to the `wireguard` package (wireguard should know how to bootstrap its port supplier). Ideally, we should keep the artificial DI layer thin, and keep things closer to where they are used. Hence I don't think there's anything to gain by moving this `portSupplier` up in the hierarchy, and it may even be wrong, considering the package architecture.

Reply:
But you're doing both now. You're sending in the port supplier which is then overriden by the concrete implementation in the port package. You either need to get rid of the injectable dependency, or inject a proper one instead.